Adds more Collision features to CSGShape3D when use_collision is enabled

bkmeneguello opened the pull request:
# Context

I'm developing an architectural visualization project where the walls have openings (doors) but must be pickable for editing, even when hovering these openings.
I'm using the CSGCombiner3D nodes to create the walls and settings `use_collision=true` to make the first-person collision, where the player must pass through the doors, but I'm also creating an Area3D matching the walls' extent to allow the picking action.

# Previous Behaviour

There were two issues when performing these actions: first, the camera pick didn't limit the pickable layers so, every physics objects were pickable. And second, despite the CSGShape3D objects creating the physics shapes, they didn't expose the picking properties and the `input_event`, `mouse_enter`, and `mouse_exit` signals.

# Solution

I copied the CollisionObject3D properties and signals to the CSGShape3D to make them compatible and also added a picking mask to the Camera3D to allow limiting the physics layers it works on.

The previous behaviour is kept so it doesn't break the compatibility.

![image](https://user-images.githubusercontent.com/1322552/213289801-69ea979c-c9af-4d5f-a2f7-8de4bda51115.png)
![image](https://user-images.githubusercontent.com/1322552/213289866-dfeadc42-a731-40bc-a554-d025efdc41ce.png)

bkmeneguello commented on scene/main/viewport.cpp:
```diff
@@ -707,28 +709,44 @@ void Viewport::_process_picking() {
-			} else {
-				physics_object_capture = ObjectID();
+			}
+			CSGShape3D *cs = Object::cast_to<CSGShape3D>(ObjectDB::get_instance(physics_object_capture));
```
Since CollisionObject3D and CSGShape3D don't share an interface that could be used here, I had to duplicate the code.

bkmeneguello commented:
@Calinou, docs fixed.

bkmeneguello commented:
@Calinou, formatting fixed

Calinou commented:
> @Calinou, formatting fixed

No need to ping people individually after applying fixes :slightly_smiling_face: 

Reviewing will happen in due time, but please note that the CSG module has no dedicated maintainers.

bkmeneguello commented:
If the CSG change could not be merged at least the camera picking layer could be split and merged separately?

bkmeneguello committed:
Adds a "pick_mask" attribute to the Camera3D

The "pick_mask" attribute allows restricting the physics layers the camera can pick.

bkmeneguello committed:
Adds ray_pickable and capture_input_on_drag attributes to CSGShape3D

Also adds the input_event, mouse_entered, and mouse_exited signals.

bkmeneguello commented:
@Calinou, is this Ok now?
